src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -70,6 +72,12 @@ private static void HandleEndObject(ref ReadStack state)
                 state.Current.JsonClassInfo.UpdateSortedPropertyCache(ref state.Current);
             }
 
+            if (state.Current.JsonClassInfo.ClassType == ClassType.Value)
```

Comment:
Any reason this is beneath the `UpdateSortedPropertyCache` call above? It seems like this error check should be first.

Reply:
Agreed.